core_backend/app/data_api/routers.py

```diff
@@ -0,0 +1,109 @@
+from datetime import datetime
+from typing import Annotated, List
+
+from fastapi import APIRouter, Depends
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import joinedload
+
+from ..auth.dependencies import authenticate_key
+from ..database import get_async_session
+from ..question_answer.models import QueryDB
+from ..users.models import UserDB
+from ..utils import setup_logger
+from .schemas import (
+    ContentFeedbackModel,
+    QueryModel,
+    QueryResponseErrorModel,
+    QueryResponseModel,
+    ResponseFeedbackModel,
+)
+
+logger = setup_logger()
+
+router = APIRouter(
+    prefix="/data-api",
+    dependencies=[Depends(authenticate_key)],
+    tags=["Data API"],
+)
+
+
+@router.get("/queries", response_model=List[QueryModel])
+async def get_queries(
+    start_date: datetime,
+    end_date: datetime,
```

Comment:
Can it be datetime or just dates?

Also, can we add examples in the docstring so users can easily understand the required format?

Reply:
It is datetime rn. I can see what happens if i accept date and datetime and convert date to datetime.
Good idea on example. I'll do that.